conference/src/test/java/com/kyriba/conference/management/api/ScheduleRestControllerIntegrationTest.java

```diff
@@ -0,0 +1,300 @@
+        .andExpect(content().contentType(APPLICATION_JSON_UTF8_VALUE))
+        .andExpect(content().json("{\"id\":7}"));
+  }
+
+
+  @Test
+  public void addPresentation_ifHallNotExistsResponseBadRequest() throws Exception
+  {
+    final String error = "Hall not found.";
+    doThrow(new LinkedEntityNotFoundException(error)).when(scheduleService)
+        .addPresentation(any(PresentationRequest.class));
+
+    mockMvc.perform(post("/api/v1/schedule/presentations")
+        .contentType(APPLICATION_JSON_UTF8_VALUE)
+        .content("{\n" +
+            "  \"hall\": 13,\n" +
+            "  \"topic\": {\n" +
+            "    \"title\": \"Concurrency\",\n" +
+            "    \"author\" : \"Brian Goetz\"\n" +
+            "  },\n" +
+            "  \"startTime\": \"" + of(9, 0).format(HHmm) + "\",\n" +
+            "  \"endTime\" : \"" + of(10, 0).format(HHmm) + "\"\n" +
+            "}"))
+        .andExpect(status().isBadRequest())
+        .andExpect(status().reason(error));
+  }
+
+
+  @Test
+  public void addPresentation_ifTopicNotCorrectResponseBadRequest() throws Exception
+  {
+    mockMvc.perform(post("/api/v1/schedule/presentations")
+        .contentType(APPLICATION_JSON_UTF8_VALUE)
+        .content("{\n" +
+            "  \"hall\": 11,\n" +
+            "  \"topic\": {\n" +
+            "    \"title\": \" \",\n" +
+            "    \"author\" : \" \"\n" +
+            "  },\n" +
+            "  \"startTime\": \" \",\n" +
+            "  \"endTime\" : \"" + of(10, 0).format(HHmm) + "\"\n" +
+            "}"))
+        .andExpect(status().isBadRequest());
+  }
+
+
+  @Test
+  public void addPresentation_ifTimeNotCorrectResponseBadRequest() throws Exception
+  {
+    final String error = "End time is before start time";
+    doThrow(new InvalidPresentationTimeException(error)).when(scheduleService)
+        .addPresentation(any(PresentationRequest.class));
+
+    mockMvc.perform(post("/api/v1/schedule/presentations")
+        .contentType(APPLICATION_JSON_UTF8_VALUE)
+        .content("{\n" +
+            "  \"hall\": 11,\n" +
+            "  \"topic\": {\n" +
+            "    \"title\": \"Concurrency\",\n" +
+            "    \"author\" : \"Brian Goetz\"\n" +
+            "  },\n" +
+            "  \"startTime\": \"" + of(10, 0).format(HHmm) + "\",\n" +
+            "  \"endTime\" : \"" + of(9, 0).format(HHmm) + "\"\n" +
+            "}"))
+        .andExpect(status().isBadRequest())
+        .andExpect(status().reason(error));
+  }
+
+
+  @Test
+  public void addPresentation_ifTimeIntersectionResponseBadRequest() throws Exception
+  {
+    final LocalTime startTime = of(9, 0);
+    final LocalTime endTime = of(10, 0);
+    doThrow(new PresentationTimeIntersectionException("Time intersection")).when(scheduleService)
+        .addPresentation(any(PresentationRequest.class));
+
+    mockMvc.perform(post("/api/v1/schedule/presentations")
+        .contentType(APPLICATION_JSON_UTF8_VALUE)
+        .content("{\n" +
```

Comment:
Usually, when the same body structure is used several times, it is moved to method with params. As an alternative, custom (not from application context) object mapper could be used to transform from map of properties.